src/Voron/Platform/Posix/PosixMemoryMapPager.cs

```diff
@@ -190,6 +192,106 @@ public override void ReleaseAllocationInfo(byte* baseAddress, long size)
             }
         }
 
+        public override void MaybePrefetchMemory(List<long> pagesToPrefetch)
+        {
+            if (Sparrow.Platform.Platform.CanPrefetch == false)
+                return; // not supported
+
+            if (pagesToPrefetch.Count == 0)
+                return;
+
+            for (int i = 0; i < pagesToPrefetch.Count; i++)
+            {
+                var ptr = (IntPtr)AcquirePagePointer(null, pagesToPrefetch[i]);
+                if (Syscall.madvise(ptr, 4 * PageSize, MAdvFlags.MADV_WILLNEED) == -1)
+                {
+                    // TODO :: ignore error ?
+                    var err = Marshal.GetLastWin32Error();
+                    PosixHelper.ThrowLastError(err);
+                }
+            }
+        }
+
+        public override void TryPrefetchingWholeFile()
+        {
+            if (Sparrow.Platform.Platform.CanPrefetch == false)
+                return; // not supported
+
+            var pagerState = PagerState;
+            for (var i = 0; i < pagerState.AllocationInfos.Length; i++)
+            {
+                var ptr = pagerState.AllocationInfos[i].BaseAddress;
+                if (Syscall.madvise(new IntPtr(ptr), (int)pagerState.AllocationInfos[i].Size, MAdvFlags.MADV_WILLNEED) == -1)
+                {
+                    // TODO :: ignore error ?
+                    var err = Marshal.GetLastWin32Error();
+                    PosixHelper.ThrowLastError(err);
+                }
+            }
+        }
+
+        public override void MaybePrefetchMemory(List<TreePage> sortedPages)
+        {
+            if (Sparrow.Platform.Platform.CanPrefetch == false)
+                return; // not supported
+
+            if (sortedPages.Count == 0)
+                return;
+
+            long lastPage = -1;
+            const int numberOfPagesInBatch = 8;
+            var sizeInPages = numberOfPagesInBatch; // OS uses 32K when you touch a page, let us reuse this
+            foreach (var page in sortedPages)
+            {
+                if (lastPage == -1)
+                {
+                    lastPage = page.PageNumber;
+                }
+
+                var numberOfPagesInLastPage = page.IsOverflow == false
```

Comment:
There is a lot of logic here that isn't related to Linux / Windows. It should be shared, and called once.